packages/query/src/utils/parse-template.ts

```diff
@@ -51,13 +52,15 @@ export function parseQueryResult(
 	}
 
 	const templateByUri = queryData.data?.templateByUri;
+	const is404 = templateByUri?.connectedNode === null;
```

Comment:
```suggestion
	// @todo templateByUri should expose a `is404` field to avoid the perf hit from connectedNode.
	const is404 = templateByUri?.connectedNode === null;
```